src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx

```diff
@@ -1,89 +1,105 @@
-                  let completed = false;
-                  if (rewardsIds && pastCampaigns) {
-                    completed = checkInclusion(pastCampaigns, rewardsIds);
-                  }
+    //todo: exclude in a dedicated helper function
+    let completed = false;
+    if (rewardsIds && pastCampaigns) {
+      completed = checkInclusion(pastCampaigns, rewardsIds);
+    }
 
-                  return (
-                    <QuestCardDetailled
-                      key={`available-mission-${index}`}
-                      active={true}
-                      title={quest?.attributes.Title}
-                      image={String(imgURL)}
-                      points={quest?.attributes.Points}
-                      link={quest?.attributes.Link}
-                      startDate={quest?.attributes.StartDate}
-                      endDate={quest?.attributes.EndDate}
-                      platformName={
-                        quest?.attributes.quests_platform?.data?.attributes
-                          ?.Name
-                      }
-                      slug={quest?.attributes.Slug}
-                      chains={chains}
-                      rewards={rewards}
-                      completed={completed}
-                      claimingIds={claimingIds}
-                      variableWeeklyAPY={
-                        quest?.attributes.Points > 0 && rewardType === 'weekly'
-                      }
-                      rewardRange={rewardRange}
-                    />
-                  );
-                })
-              : Array.from({ length: 3 }, () => 42).map((_, idx) => (
-                  <QuestCardSkeleton key={'mission-card-skeleton-' + idx} />
-                ))}
-          </CarouselContainer>
-        </QuestCarouselContainer>
-      ) : null}
-    </>
+    return (
+      <QuestCardDetailled
+        key={`available-mission-${index}`}
+        active={true}
+        title={quest?.attributes.Title}
+        image={String(imgURL)}
+        points={quest?.attributes.Points}
+        link={quest?.attributes.Link}
+        startDate={quest?.attributes.StartDate}
+        endDate={quest?.attributes.EndDate}
+        platformName={quest?.attributes.quests_platform?.data?.attributes?.Name}
+        slug={quest?.attributes.Slug}
+        chains={chains}
+        rewards={rewards}
+        completed={completed}
+        claimingIds={claimingIds}
+        variableWeeklyAPY={
+          quest?.attributes.Points > 0 && rewardType === 'weekly'
+        }
+        rewardRange={rewardRange}
+      />
+    );
+  };
+
+  return (
+    !isNotLive && (
+      <QuestCarouselContainer>
+        <CarouselContainer title={t('missions.available')}>
+          {loading ? (
+            Array.from({ length: 3 }, (_, idx) => (
+              <QuestCardSkeleton key={`mission-card-skeleton-${idx}`} />
+            ))
+          ) : (
+            <>
+              {quests?.map((quest: Quest, index: number) =>
```

Comment:
not sure about this logic, feels weird. @oktapodia any viewpoint?

Reply:
My render-functions were an attempt to split logic from render function. I am still not sure what´s best, but I removed those functions for now.